src/Sql/Vault/dbo/Stored Procedures/Cipher/Cipher_SoftDelete_V2.sql

```diff
@@ -0,0 +1,60 @@
+﻿CREATE PROCEDURE [dbo].[Cipher_SoftDelete_V2]
+    @Ids AS [dbo].[GuidIdArray] READONLY,
+    @UserId AS UNIQUEIDENTIFIER
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    CREATE TABLE #Temp
+    (
+        [Id] UNIQUEIDENTIFIER NOT NULL,
+        [UserId] UNIQUEIDENTIFIER NULL,
+        [OrganizationId] UNIQUEIDENTIFIER NULL
+    )
+
+    INSERT INTO #Temp
+    SELECT
+        [Id],
+        [UserId],
+        [OrganizationId]
+    FROM
+        [dbo].[UserCipherDetails_V2](@UserId)
+    WHERE
+        [Edit] = 1
+        AND [DeletedDate] IS NULL
+        AND [Id] IN (SELECT * FROM @Ids)
+
```

Comment:
Possible to avoid in:
INSERT INTO #Temp
SELECT
    UCDV2.[Id],
    UCDV2.[UserId],
    UCDV2.[OrganizationId]
FROM
    [dbo].[UserCipherDetails_V2](@UserId) UCDV2
INNER JOIN
    @Ids I
ON
    UCDV2.[Id] = I.Id
WHERE
    UCDV2.[Edit] = 1
    AND UCDV2.[DeletedDate] IS NULL